packages/ibm-products/src/components/Toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright IBM Corp. 2021, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import cx from 'classnames';
+import { bool, node, string } from 'prop-types';
+
+import React, {
+  PropsWithChildren,
+  ReactNode,
+  createContext,
+  forwardRef,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { getDevtoolsProps } from '../../global/js/utils/devtools';
+// cspell:words Focusable focusable
+import { getFocusableElements as _getFocusableElements } from '../../global/js/utils/getFocusableElements';
+import { pkg } from '../../settings';
+
+const { checkComponentEnabled, prefix } = pkg;
+const blockClass = `${prefix}--toolbar`;
+
+interface ToolbarProps {
+  /** Provide the content of the `ToolbarGroup` */
+  children: ReactNode;
+
+  /** Provide an optional class to be applied to the containing node */
+  className?: string;
+
+  /** Determines whether the `Toolbar` is rendered vertically */
+  vertical?: boolean;
+}
+
+interface ToolbarContextType {
+  vertical?: boolean;
+}
+
+const ToolbarContext = createContext<ToolbarContextType>({});
+
+/** Toolbars are a collection of action items that organize a program’s interaction patterns into a series of closely related commands. */
+let Toolbar = forwardRef(
+  (
+    { children, className, vertical, ...rest }: PropsWithChildren<ToolbarProps>,
+    r: React.Ref<any>
+  ) => {
+    const focusableElements = useRef();
+
+    const getFocusableElements = useCallback(
+      (): any => focusableElements.current,
+      [focusableElements]
+    );
+
+    const _ref = useRef<any>();
+    const ref = r || _ref;
```

Comment:
In these cases where we use the forwarded ref or a local one we do something like this:
```jsx
const localRef = useRef<HTMLDivElement>();
const ref = ref || localRef;
```

I know this is more of a naming thing but would be good to keep a consistent approach across components.

Reply:
Renamed as you advised